### PR TITLE
BUG: add sorting_reamaining=True to avoid IndexError GH25831

### LINKED ISSUE #25831: Can't sort DataFrame when level=[] and sort_remaining=False
#### Code Sample

```python
import pandas as pd

# Create a DataFrame with a MultiIndex.
df = pd.DataFrame({
    "a": [0, 2, 1],
    "b": [3, 0, 2],
    "c": [0, 0, 1]
}).set_index(["a", "b"])

# This works.
df.sort_index(level=[], sort_remaining=True)

# This throws an IndexError: list index out of range. 
df.sort_index(level=[], sort_remaining=False)
```

#### Problem description

Pandas is unable to sort the index of a DataFrame when level=[] and sort_remaining=False. It works when sort_remaining=True, though. This is inconsistent behavior; it doesn't make sense for one to work but not the other.

The above code worked as intended in 0.22, but started throwing an error in 0.23. I believe that the commit that broke it was [this one](https://github.com/pandas-dev/pandas/commit/bc37ea2e05019a89adaa48159b220483598d1898#diff-1e79abbbdd150d4771b91ea60a4e1cc7). By changing 
```
if level:
```
to 
```
if level is not None:
```
the conditional is now entered when `level=[]`, whereas it wasn't previously (since an empty list is treated as `False`).

#### Expected Output

The offending line of code should be equivalent to `df`.

(You might wonder how the above situation would even arise; no one would ever deliberately write the above line of code, since it is a no-op. But it can happen when the `level=` argument is being programmatically generated.)

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.8.final.0
python-bits: 64
OS: Darwin
OS-release: 16.5.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.1
pytest: 3.3.2
pip: 18.1
setuptools: 38.4.0
Cython: 0.27.3
numpy: 1.15.4
scipy: 1.2.1
pyarrow: None
xarray: None
IPython: 6.2.1
sphinx: 1.6.6
patsy: 0.5.0
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.9
feather: None
matplotlib: 2.1.2
openpyxl: 2.4.10
xlrd: 1.1.0
xlwt: 1.2.0
xlsxwriter: 1.0.2
lxml.etree: 4.1.1
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.1
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by YuRenee**
- [x] closes #25831
- [x] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by arw2019 — COMMENTED** [review]
Can you add test(s)? It's the first thing we look for when reviewing

**Comment by YuRenee** [comment]
Sure!

**Comment by YuRenee** [comment]
Just added!

File: pandas/core/sorting.py [hunk 1]
```diff
@@ -68,7 +68,11 @@ def get_indexer_indexer(
     target = ensure_key_mapped(target, key, levels=level)
     target = target._sort_levels_monotonic()
 
-    if level is not None:
+    if isinstance(level, (str, int)) and level is not None:
```
Comment (jreback) [inline pandas/core/sorting.py:71]: you don't need the not None conditions

File: pandas/core/sorting.py [hunk 2]
```diff
@@ -68,7 +68,11 @@ def get_indexer_indexer(
...
+        _, indexer = target.sortlevel(
+            level, ascending=ascending, sort_remaining=sort_remaining
+        )
+    elif isinstance(level, list) and len(level) != 0:
```
Comment (jreback) [inline pandas/core/sorting.py:75]: don't use list testing directly rather is_list_like

File: pandas/core/sorting.py (hunk 1 above)
Comment (jreback) [reply pandas/core/sorting.py:71]: use is_scalar(level) instead

File: pandas/core/sorting.py (hunk 2 above)
Comment (jreback) [reply pandas/core/sorting.py:75]: rather just make this robust to an empty level

File: pandas/core/sorting.py [hunk 3]
```diff
@@ -446,12 +450,11 @@ def _ensure_key_mapped_multiindex(
         Resulting MultiIndex with modified levels.
     """
 
-    if level is not None:
-        if isinstance(level, (str, int)):
-            sort_levels = [level]
-        else:
-            sort_levels = level
-
+    if isinstance(level, (str, int)) and level is not None:
```
Comment (jreback) [inline pandas/core/sorting.py:453]: same comments as above

File: pandas/tests/frame/methods/test_sort_index.py [hunk 4]
```diff
@@ -857,3 +857,14 @@ def test_sort_index_multiindex_sparse_column(self):
...
+
+    def test_sort_index_sorting_remaining(self):
+        # GH#25831
+        # Create a DataFrame with a MultiIndex.
+        df = DataFrame({"a": [0, 2, 1], "b": [3, 0, 2], "c": [0, 0, 1]}).set_index(
+            ["a", "b"]
+        )
+
+        result = df.sort_index(level=[], sort_remaining=False)
+        expected = df.sort_index(level=[], sort_remaining=True)
```
Comment (jreback) [inline pandas/tests/frame/methods/test_sort_index.py:869]: an you hard-code the expected result frame.

can you parameterize this test for sort_remaining =True/False

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you merge master

File: pandas/core/indexes/category.py [hunk 5]
```diff
@@ -324,7 +324,9 @@ def _format_attrs(self):
                 "categories",
                 ibase.default_pprint(self.categories, max_seq_items=max_categories),
             ),
-            ("ordered", self.ordered),
+            # pandas\core\indexes\category.py:315: error: "CategoricalIndex"
```
Comment (jreback) [inline pandas/core/indexes/category.py:327]: hmm CI shouldn't have an attribute ordered, rather the CDT does

File: pandas/core/indexes/category.py [hunk 6]
```diff
@@ -659,6 +661,14 @@ def map(self, mapper):
 
     def _concat(self, to_concat: List["Index"], name: Label) -> "CategoricalIndex":
         # if calling index is category, don't check dtype of others
+        count = 0
+        for c in to_concat:
+            if not to_concat[0].ordered:
```
Comment (jreback) [inline pandas/core/indexes/category.py:666]: can you do a list comprehension instead,using counters like this is not pythonic

File: pandas/core/reshape/concat.py [hunk 7]
```diff
@@ -502,12 +501,12 @@ def get_result(self):
                     obj_labels = obj.axes[1 - ax]
                     if not new_labels.equals(obj_labels):
                         indexers[ax] = obj_labels.reindex(new_labels)[1]
-
                 mgrs_indexers.append((obj._mgr, indexers))
 
             new_data = concatenate_block_managers(
                 mgrs_indexers, self.new_axes, concat_axis=self.bm_axis, copy=self.copy
             )
+
```
Comment (jreback) [inline pandas/core/reshape/concat.py:509]: ideally if you dont' touch whitespace would be good

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you merge master and update

**Comment by jreback** [comment]
closing as stale. if you want to continue, pls ping and can re-open.

**Pull request closed without merging**